Source/CarthageKit/Project.swift

```diff
@@ -960,6 +960,22 @@ public final class Project {
 			}
 	}
 
+	public func generateSwiftPackageManagerXcodeProjectIfAvailable(forDependencyAt dependencyPath: String) -> SignalProducer<(), NoError> {
+		let files = (try? FileManager.default.contentsOfDirectory(atPath: dependencyPath)) ?? []
+
+		let hasXcodeProjectFile = files.lazy.filter { $0.hasSuffix(".xcodeproj") || $0.hasSuffix(".xcworkspace") }.first != nil
```

Comment:
Instead of doing `foo.lazy.filter { ... }.first`, you should be able to do `foo.first { ... }`.